### PR TITLE
[Refactor](ScanNode) Split interface refactor.

### PR DESCRIPTION
**Pull request by Jibing-Li**
Move getSplits function to ScanNode, remove Splitter interface.
For each kind of data source, create a specific ScanNode and implement the getSplits interface. For example, HiveScanNode.
Remove FileScanProviderIf move the code to each ScanNode.

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.79 seconds
 stream load tsv:          426 seconds loaded 74807831229 Bytes, about 167 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230428081524_clickbench_pr_137027.html

**Comment by Jibing-Li** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/spi/Split.java [hunk 1]
```diff
@@ -15,17 +15,15 @@
...
+ */
+public interface Split {
 
-@Data
-public abstract class Split {
-    protected String[] hosts;
+    String[] getHosts();
 
-    public Split() {}
+    Object getInfo();
```
Comment (xxiao2018) [inline fe/fe-core/src/main/java/org/apache/doris/spi/Split.java:27]: What is this for?

Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/spi/Split.java:27]: There may be significant variance between different splits, this reserved interface may may useful in the future. But for now, it is useless.

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**